xsdata/transformer.py

```diff
@@ -23,71 +23,97 @@ class SchemaTransformer:
     print: bool
     format: str
     processed: List[Path] = field(init=False, default_factory=list)
-    included: List[Path] = field(init=False, default_factory=list)
 
-    def process(
+    def process(self, schema_path: Path, package: str):
+        classes = self.process_schema(schema_path, package)
+        classes = self.analyze_classes(classes)
+        class_num, inner_num = self.count_classes(classes)
+
+        if not class_num:
+            return logger.warning("Analyzer returned zero classes!")
+
+        logger.info("Analyzer: %d main and %d inner classes", class_num, inner_num)
+
+        writer.designate(classes, self.format)
+        if self.print:
+            writer.print(classes, self.format)
+        else:
+            writer.write(classes, self.format)
+
+    def analyze_classes(self, classes: List[Class]):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5141157)